lib/model/avatar_url.dart

```diff
@@ -0,0 +1,141 @@
+import 'dart:convert';
+
+import 'package:crypto/crypto.dart';
+
+import '../log.dart';
+
+/// Pixel dimensions of different size choices we have (they're all
+/// square) when requesting an uploaded avatar.
+/// DEFAULT_AVATAR_SIZE in zerver/lib/upload.py.
+const int defaultUploadSizePx = 100;
+
+/// MEDIUM_AVATAR_SIZE in zerver/lib/upload.py.
+const int mediumUploadSizePx = 500;
+
+abstract class AvatarUrl {
+  static AvatarUrl fromUserData({
```

Comment:
Let's make this a factory constructor too.

In fact it can be named as just the default constructor:
```suggestion
  factory AvatarUrl({
```

(That way will require a couple of small fixes for the sake of the subclasses' constructors. If you're having trouble working those out, you can make it `factory AvatarUrl.fromUserData`.)

Reply:
[ef7582d](https://github.com/zulip/zulip-flutter/pull/534/commits/ef7582dc24572904bd186034795977cec0e0886a)